### PR TITLE
Use WASM pull secret if protected registry

### LINKED ISSUE #1077: Support protected registry when loading WASM image
**Describe the bug**
When loading the WASM image we provide the registry URL to Isito via the WASMPlugin resource. However we do not have a way to configure a imagePullSecret to authenitcate against a registry that is protected. 

**To Reproduce**

- Delete any existing policies
- Set the RELATED_IMAGE_WASMSHIM in the operator subscription to one behind a protected registry
 - Create a new policy
 - Watch the logs of Istio it will fail to load the image as it is unauthenticated.

**Expected behavior**
- Provide a way to accept a imagePullSecret that we could then pass on to the WASMPlugin resource created in response to a policy being created.

**Additional context**
[Manual Workaround.](https://gist.github.com/eguzki/501635c4a4d2161cd188fe4b210d77b0)

### PR DESCRIPTION
**Pull request by maleck13**
closes #1077 
Signed-off-by: craig <cbrookes@redhat.com>

rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

### CODE REVIEW

File: controllers/state_of_the_world.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ import (
 
 var (
 	operatorNamespace       = env.GetString("OPERATOR_NAMESPACE", "kuadrant-system")
+	wasmImagePullSecret     = env.GetString("WASM_IMAGE_PULL_SECRET", "")
```
Comment (maleck13) [inline controllers/state_of_the_world.go:46]: going to move this into the istio_extenstion_reconciler

**Comment by maleck13** [comment]
adding an update to the OpenShfit install doc

**Comment by eguzki** [comment]
Have you considered doing the same for EnvoyGateway (in the [EnvoyExtensionPolicy](https://gateway.envoyproxy.io/docs/api/extension_types/#imagewasmcodesource))? (Not asking to do so...)

**Comment by maleck13** [comment]
@trepel FYI pull secret update

File: controllers/data_plane_policies_workflow.go [hunk 2]
```diff
@@ -25,6 +25,11 @@ const (
 
 var (
 	WASMFilterImageURL = env.GetString("RELATED_IMAGE_WASMSHIM", "oci://quay.io/kuadrant/wasm-shim:latest")
+	// protectedRegistry this defines a default protected registry. If this is in the wasm image URL we add a pull secret name to the WASMPLugin resource
+	ProtectedRegistry = env.GetString("PROTECTED_REGISTRY", "registry.redhat.com")
```
[the remark this answers is not in the document]
Comment (maleck13) [reply controllers/data_plane_policies_workflow.go:29]: good spot!

Comment (maleck13) [inline controllers/data_plane_policies_workflow.go:29]:
```suggestion
	ProtectedRegistry = env.GetString("PROTECTED_REGISTRY", "registry.redhat.io")
```

**Comment by eguzki** [comment]
Verification steps working like a charm for Istio and EnvoyGateway :medal_military:

**Review by eguzki — APPROVED** [review]
LGTM

**Comment by eguzki** [comment]
@maleck13  One thing I found, worth mentioning, is that the `imagePullSecret` field is not reconciled, so if the operator is restarted with a new wasm shim image from a "public" registry (like `https://quay.io/repository/kuadrant/wasm-shim`), the  `imagePullSecret` field is not set to empty or null.

**Comment by eguzki** [comment]
Updated the verification steps to cover the reconciliation when the url changes

**Review by eguzki — APPROVED** [review]
LGTM

Great work :medal_military:

**Pull request merged**